optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -390,6 +383,9 @@ public static int resolvePoolSize(String propertyName, String value, String... m
                 typeArgument = upperBounds[0];
             }
         }
+        if (typeArgument instanceof ParameterizedType) {
+            return (Class) ((ParameterizedType) typeArgument).getRawType();
+        }
```

Comment:
Another instance of an over-eager fail-fast. `Collection<Something<T>>` shouldn't fail just because we don't know what `T` is. It's a `Collection<Something>` then.

Reply:
This the same code as on line 375/365? So that code is already there?